src/renderer/screens/exchange/Swap2/Form/index.js

```diff
@@ -32,19 +39,37 @@ const SwapForm = () => {
   // SWAP MOCK - PLEASE REMOVE ME ASA LOGIC IS IMPLEMENTED
   const [isSwapReady] = useState(false);
   const { t } = useTranslation();
+  const { providers, error } = useSwapProviders();
+  const dispatch = useDispatch();
+  const storedProviders = useSelector(providersSelector);
 
   // SWAP MOCK - PLEASE REMOVE ME ASA LOGIC IS IMPLEMENTED
   const onSubmit = () => {};
 
-  return (
-    <Wrapper>
-      <SwapFormSelectors />
-      <SwapFormSummary {...summaryMockedData} />
-      <Button primary disabled={!isSwapReady} onClick={onSubmit}>
-        {t("common.exchange")}
-      </Button>
-    </Wrapper>
-  );
+  useEffect(() => {
+    if (providers) dispatch(updateProvidersAction(providers));
+  }, [providers]);
+
+  useEffect(() => {
+    if (error) dispatch(resetSwapAction());
+  }, [error]);
+
+  if (providers?.length)
+    return (
+      <Wrapper>
+        <SwapFormSelectors />
+        <SwapFormSummary {...summaryMockedData} />
+        <Button primary disabled={!isSwapReady} onClick={onSubmit}>
+          {t("common.exchange")}
+        </Button>
+      </Wrapper>
+    );
+
+  // TODO: ensure that the error is catch by Sentry in this case
+  if (error) return <FormNotAvailable />;
```

Comment:
use the Track component to track the event, and unify those into one if they are the same

Reply:
Do we wanna track page states (loading/error)? 🤔 
Navbar component already sends analytics based on the route, using the `TrackPage` component. 

```js
// from src/renderer/screens/exchange/Swap2/Navbar/index.js
<Nav>
      <TrackPage category="swap" name={swapRoutes[currentIndex].name} />
      <TabBar tabs={tabs} onIndexChange={onWrappedTabChange} index={currentIndex} />
    </Nav>
``` 
However it only sends routing information but nothing related to the state of the page is send (the page is loaded, the page displays the error state).
Do we want this behaviour for metrics usage? If yes, do we have a way to send events programmatically to add more granularity to the last navigation event sent or should I considered loading/error states as "sub-pages" and add a particular suffix to the form page name? For example:

```js
<TrackPage category="swap" name={history} /> // from history page
<TrackPage category="swap" name={form} /> // from form page
<TrackPage category="swap" name={form:loading} /> // from form page when the form is loading
<TrackPage category="swap" name={form:error} /> // from form page when the form is displaying the error state
```

Reply:
`TrackPage` for screen loads, and the `Track` for events inside of them. 
Track also takes a few props to determine when they get triggered, so they can only load on render, or on unmount, or whatever, maybe they match your needs. I'd say go with something that resembles the events we already have, and then we will need to sync with product on what exactly they want tracked

Reply:
Perfect, I added `Track` component using the same event name that you used from the last version of the swap.